### PR TITLE
bump 3dt

### PR DESCRIPTION
**Pull request by darkonie**
## High Level Description

3DT bug fixes, refactoring and improvements.

## Related Issues

  # Issues

[DCOS-11680](https://mesosphere.atlassian.net/browse/DCOS-11680)
[DCOS-11776](https://mesosphere.atlassian.net/browse/DCOS-11776)
[DCOS-11319](https://mesosphere.atlassian.net/browse/DCOS-11319)
[DCOS-11681](https://mesosphere.atlassian.net/browse/DCOS-11681)
[DCOS-9716](https://mesosphere.atlassian.net/browse/DCOS-9716)

## Checklist for all PR's

  - [X] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [X] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [X] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [X] Change log from the last version integrated (this should be a link to commits for easy verification and review): [3DT](https://github.com/dcos/3dt/compare/c9a7755fa75bd1c96192a26daa6a6fea0c714001...0c0ffc7d66e6396d64723ad26f7615164edfe3ce)
  - [X] Test Results: [285](https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-3dt-pulls/285/console)
  - [X] Code Coverage (if available): [285](https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-3dt-pulls/285/console)
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by malnick** [comment]
Nice. I made tickets to make sure we update role discovery for Metrics and Signal service as well.

**Comment by darkonie** [comment]
@malnick @adam-mesos @lingmann this is ready for review

File: packages/dcos-integration-test/extra/test_3dt.py [hunk 1]
```diff
@@ -496,6 +498,22 @@ def test_3dt_bundle_download_and_extract(dcos_api_session):
...
+    """
+    test bundle download and validate zip file
+    """
+    if len(dcos_api_session.masters) < 3:
+        pytest.skip('Test requires at least 3 master nodes')
+
+    _download_bundle(dcos_api_session, master_index=1)
+
+
+def _download_bundle(dcos_api_session, master_index=0):
```
Comment (mellenburg) [inline packages/dcos-integration-test/extra/test_3dt.py]: Please make master_index a regular required arg. It seems a little confusing as a default. And perhaps call this function '_download_bundle_from_master'.

Comment (darkonie) [inline packages/dcos-integration-test/extra/test_3dt.py]: done

**Comment by adammesos** [comment]
CI is actually green if you follow the links. @malnick Can we get your approval on the 3dt-specifics, and then we can pull this into the next train?

**Comment by malnick** [comment]
:ship:

**Comment by amitaekbote** [comment]
Do we need to make any specific changes to enterprise for this pr? Tests?

**Comment by darkonie** [comment]
@amitaekbote yes, you can cherry pick these commits https://github.com/mesosphere/dcos-enterprise/pull/514

**Pull request merged**